### PR TITLE
fix cardano send max and spendable balance

### PR DESCRIPTION
**Pull request by hedi-edelbloute**
### 📝 Description

Fix cardano spendable balance on LLD as it included twice the reward
Fix cardano amount on last step before signing

Integration tests fail the same way on develop : https://github.com/LedgerHQ/ledger-live/actions/runs/6967769829/job/18960305105

Adding unit test for the utxo sum is really difficult as I don't know the specifics on utxos/internalcredential/paymentCred thing

### ❓ Context

- **JIRA or GitHub link**: [https://ledgerhq.atlassian.net/browse/LIVE-9913](https://ledgerhq.atlassian.net/browse/LIVE-9913), [https://ledgerhq.atlassian.net/browse/LIVE-10253](https://ledgerhq.atlassian.net/browse/LIVE-10253)
### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - Send max amount on confirmation with device screen
  - Available balance

---

### 🧐 Checklist for the PR Reviewers

- [] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [] **Any new dependencies** have been justified and documented.
- [] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/ledger-live-common/src/families/cardano/js-synchronisation.ts [hunk 1]
```diff
@@ -295,7 +295,7 @@ const makeGetAccountShape =
     );
 
     const utxos = prepareUtxos(newTransactions, stableUtxos, accountCredentialsMap);
-    let accountBalance = utxos.reduce((total, u) => total.plus(u.amount), new BigNumber(0));
+    let utxosSum = utxos.reduce((total, u) => total.plus(u.amount), new BigNumber(0));
```
Comment (chabroA) [inline libs/ledger-live-common/src/families/cardano/js-synchronisation.ts:298]:
```suggestion
    const utxosSum = utxos.reduce((total, u) => total.plus(u.amount), new BigNumber(0));
```

cf. https://github.com/LedgerHQ/ledger-live/actions/runs/6967233140/job/18958758900#step:6:299

File: libs/ledger-live-common/src/families/cardano/js-synchronisation.ts [hunk 2]
```diff
@@ -349,8 +352,8 @@ const makeGetAccountShape =
     return {
       id: accountId,
       xpub,
-      balance: accountBalance,
-      spendableBalance: accountBalance.minus(minAdaBalanceForTokens),
+      balance: totalBalance,
```
Comment (chabroA) [inline libs/ledger-live-common/src/families/cardano/js-synchronisation.ts:355]: I would get rid of the extra `totalBalance` variable above.

```suggestion
      balance: delegationInfo?.rewards ? utxosSum.plus(delegationInfo.rewards) : utxosSum,
```

File: libs/ledger-live-common/src/families/cardano/js-synchronisation.ts [hunk 3]
```diff
@@ -320,8 +320,11 @@ const makeGetAccountShape =
       }));
     const cardanoNetworkInfo = await getNetworkInfo(initialAccount as CardanoAccount, currency);
     const delegationInfo = await getDelegationInfo(currency, stakeCredential.key);
+
+    let totalBalance = utxosSum;
+
     if (delegationInfo?.rewards) {
-      accountBalance = accountBalance.plus(delegationInfo.rewards);
+      totalBalance = utxosSum.plus(delegationInfo.rewards);
```
Comment (chabroA) [inline libs/ledger-live-common/src/families/cardano/js-synchronisation.ts:327]: I would get rid of the `totalBalance` variable (cf. other comment below).

Alternatively, you can keep it but simplify it's definition with:
```ts
const totalBalance = delegationInfo?.rewards ? utxosSum.plus(delegationInfo.rewards) : utxosSum
```

And keep using it below in the return object as:
```ts
return {
  ...
  balance: totalBalance,
  ...
}

```

**Review by chabroA — CHANGES_REQUESTED** [review]

File: libs/ledger-live-common/src/families/cardano/js-synchronisation.ts (hunk 3 above)
Comment (hedi-edelbloute) [reply libs/ledger-live-common/src/families/cardano/js-synchronisation.ts]: Done

File: libs/ledger-live-common/src/families/cardano/js-synchronisation.ts (hunk 2 above)
Comment (hedi-edelbloute) [reply libs/ledger-live-common/src/families/cardano/js-synchronisation.ts:351]: Done

File: libs/ledger-live-common/src/families/cardano/js-synchronisation.ts (hunk 1 above)
Comment (hedi-edelbloute) [reply libs/ledger-live-common/src/families/cardano/js-synchronisation.ts]: Done

**Review by chabroA — APPROVED** [review]

**Pull request merged**